shared/feature/src/commonTest/kotlin/app/futured/kmptemplate/feature/ui/firstScreen/FirstComponentTest.kt

```diff
@@ -0,0 +1,237 @@
+package app.futured.kmptemplate.feature.ui.firstScreen
+
+import app.cash.turbine.Turbine
+import app.cash.turbine.turbineScope
+import app.futured.arkitekt.crusecases.error.UseCaseErrorHandler
+import app.futured.kmptemplate.feature.domain.CounterUseCase
+import app.futured.kmptemplate.feature.domain.FetchDataUseCase
+import app.futured.kmptemplate.feature.navigation.home.HomeConfig
+import app.futured.kmptemplate.feature.ui.base.DefaultAppComponentContext
+import app.futured.kmptemplate.network.rest.dto.Person
+import app.futured.kmptemplate.network.rest.result.NetworkError
+import com.arkivanov.decompose.DefaultComponentContext
+import com.arkivanov.essenty.lifecycle.LifecycleRegistry
+import com.arkivanov.essenty.lifecycle.create
+import com.arkivanov.essenty.lifecycle.destroy
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOf
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.UnconfinedTestDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import kotlinx.io.IOException
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+import kotlin.time.Duration.Companion.seconds
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class FirstComponentTest {
+
+    private class FakeNavigation : FirstScreenNavigation {
+        val calls = Turbine<HomeConfig>()
+
+        override fun FirstComponent.navigateToSecond() {
+            calls.add(HomeConfig.Second)
+        }
+    }
+
+    private val testScope = TestScope()
+    private val testDispatcher = UnconfinedTestDispatcher(testScope.testScheduler)
+
+    private val lifecycleRegistry = LifecycleRegistry()
+    private val appComponentContext = DefaultAppComponentContext(DefaultComponentContext(lifecycleRegistry))
+
+    @BeforeTest
+    fun setup() {
+        Dispatchers.setMain(testDispatcher)
+        UseCaseErrorHandler.globalOnErrorLogger = {}
+    }
+
+    @AfterTest
+    fun cleanup() {
+        lifecycleRegistry.destroy()
+        Dispatchers.resetMain()
+    }
```

Comment:
@BeforeTest and @AfterTest will repeat a lot in these types of tests. It would be better to create something like 

```
object ComponentTestPreparation {

    fun setup(...) { ....} 
    
    ....
    
}
```